can/io/gz.py

```diff
@@ -0,0 +1,37 @@
+"""
+This module contains handling for .gz (gzip) files.
+"""
+
+import gzip
+import pathlib
+import tempfile
+
+from ..typechecking import AcceptedIOType
+
+
+class GzipReader:
+    """
+    Handle any supported file that may be compressed using gzip.
+    Decompresses the file and transfers contents to a tempfile.
+    The tempfile then gets re-consumed by LogReader and automatically
+    removed once complete.
+    """
+
+    @staticmethod
+    def __new__(cls, file: AcceptedIOType):
+        """
+        :param file: a path-like object or as file-like object to read from.
+        """
+        # Needed to avoid circular import error
+        from .player import LogReader
+
+        real_suffix = pathlib.Path(file).suffixes[-2]
+        file_mode = "b" if real_suffix in [".blf"] else "t"
+
+        with tempfile.NamedTemporaryFile(
+            mode=f"w{file_mode}", suffix=real_suffix
```

Comment:
I think for the copy procedure, we do not need to decode and re-encode the file, we can stay in binary `file_mode`, right?

Reply:
you're right!

I was experimenting with using the `io` module which required I select between `StringIO` or `BytesIO` -- if a `tempfile` is used it's not needed since we don't need to read the file and can use something like `shutil` to copy.